specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -2071,86 +3243,249 @@
+      ],
+      "type": "object",
+      "properties": {
+        "keyIdentifier": {
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9_]",
+          "type": "string"
+        }
+      }
+    },
+    "EncryptionProperty": {
+      "required": [
+        "keyVaultProperties",
+        "status"
+      ],
+      "type": "object",
+      "properties": {
+        "cosmosDbResourceId": {
+          "description": "The byok cosmosdb account that customer brings to store customer's data\r\nwith encryption",
+          "type": "string"
+        },
+        "identity": {
+          "description": "Identity to be used with the keyVault",
+          "$ref": "#/definitions/IdentityForCmk"
+        },
+        "keyVaultProperties": {
+          "description": "KeyVault details to do the encryption",
+          "$ref": "#/definitions/KeyVaultProperties"
+        },
+        "searchAccountResourceId": {
+          "description": "The byok search account that customer brings to store customer's data\r\nwith encryption",
+          "type": "string"
+        },
+        "status": {
+          "$ref": "#/definitions/EncryptionStatus"
+        },
+        "storageAccountResourceId": {
+          "description": "The byok storage account that customer brings to store customer's data\r\nwith encryption",
+          "type": "string"
+        }
+      }
+    },
+    "EncryptionStatus": {
+      "description": "Indicates whether or not the encryption is enabled for the workspace.",
+      "enum": [
+        "Enabled",
+        "Disabled"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "EncryptionStatus",
+        "modelAsString": true
+      }
+    },
+    "EncryptionUpdateProperties": {
+      "required": [
+        "keyVaultProperties"
+      ],
+      "type": "object",
+      "properties": {
+        "keyVaultProperties": {
+          "$ref": "#/definitions/EncryptionKeyVaultUpdateProperties"
+        }
+      }
+    },
+    "EndpointDeploymentModel": {
+      "type": "object",
+      "properties": {
+        "format": {
+          "description": "Model format",
+          "type": "string"
+        },
+        "name": {
+          "description": "Model name.",
+          "type": "string"
+        },
+        "source": {
+          "description": "Optional. Deployment model source ARM resource ID.",
+          "type": "string"
+        },
+        "version": {
+          "description": "Model version.",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointDeploymentResourceProperties": {
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "properties": {
+        "failureReason": {
+          "description": "The failure reason if the creation failed.",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Read-only provision state status property.",
+          "$ref": "#/definitions/DefaultResourceProvisioningState"
+        },
+        "type": {
```

Comment:
>type [](http://example.com/codeflow?start=9&length=4)

Rename to avoid duplicating top-level ARM property. Also, please use a string enum for discriminator value.

Reply:
Same as comment above, we wish to migrate some existing API so hope we can keep this schema as it.